binding.cc

```diff
@@ -1179,6 +1180,29 @@ namespace zmq {
     NanReturnValue(NanNew<String>(version_info));
   }
 
+#if ZMQ_VERSION_MAJOR >= 4
+   static NAN_METHOD(ZmqCurveKeypair) {
+    NanEscapableScope();
+
+    char public_key [41];
+    char secret_key [41];
+
+    int rc = zmq_curve_keypair( public_key, secret_key);
+    if (rc < 0) {
+      throw std::runtime_error(ErrorMessage());
+    }
+
+    Persistent<Object> persistentHandle;
```

Comment:
Why do you create a `Persistent` handle here? It will not be in visible in scope for anything else, so you can never refer to it outside this function.